app/grandchallenge/github/tasks.py

```diff
@@ -47,27 +46,73 @@ def get_zipfile(*, pk):
         timeout=10,
     )
     access_token = json.loads(resp.content)["token"]
-    full_name = payload["repository"]["full_name"]
-    headers["Authorization"] = f"token {access_token}"
-    zipfile_url = (
-        f"https://api.github.com/repos/{full_name}/zipball/{payload['ref']}"
-    )
-    with requests.get(
-        zipfile_url, headers=headers, timeout=10, stream=True
-    ) as file:
-        with NamedTemporaryFile(delete=True) as tmp_file:
-            with open(tmp_file.name, "wb") as fd:
-                for chunk in file.iter_content(chunk_size=128):
-                    fd.write(chunk)
 
-            tmp_file.flush()
-            temp_file = files.File(
-                tmp_file, name=f"{ghwm.repo_name}-{ghwm.tag}.zip",
-            )
+    repo_url = payload["repository"]["html_url"]
+    return repo_url.replace("//", f"//x-access-token:{access_token}@")
 
-            ghwm.zipfile = temp_file
-            ghwm.save()
 
+@shared_task()
+def get_zipfile(*, pk):
+    GitHubWebhookMessage = apps.get_model(  # noqa: N806
+        app_label="github", model_name="GitHubWebhookMessage"
+    )
+    ghwm = GitHubWebhookMessage.objects.get(pk=pk)
+    payload = ghwm.payload
+    repo_url = get_repo_url(payload)
+    zip_name = f"{ghwm.repo_name}-{ghwm.tag}.zip"
+    tmp_zip = tempfile.NamedTemporaryFile()
+    has_open_source_license = False
+    license = "No license file found"
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        proces = subprocess.Popen(
+            [
+                "git",
+                "clone",
+                "--recurse-submodules",
```

Comment:
Would be nice to make recursing submodules an option as sometimes this can lead to issues (eg, for a repo we can't access but which is unimportant for building the inference container).

Reply:
I guess that would then be an option on the algorithm object? I agree that would be a smart thing to do, as it could potentially lower the zip size by a substantial amount and like you said, we might run into authentication issues with the submodules.